src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -175,20 +198,26 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
   }, [itemToShare, roles]);
 
   const loadShareInfo = async () => {
+    if (!itemToShare?.item) return;
+
     setIsLoading(true);
     // Change object type of itemToShare to AdvancedSharedItem
     let shareAccessMode: AccessMode = 'public';
     let sharingType = 'public';
+    let isAlreadyPasswordProtected = false;
 
-    if (props.isDriveItem) {
-      sharingType = (itemToShare?.item as DriveItemData & { sharings: { type: string; id: string }[] }).sharings?.[0]
-        ?.type;
-    } else {
-      const itemType = itemToShare?.item.isFolder ? 'folder' : 'file';
-      const itemId = itemToShare?.item.uuid ?? '';
+    const itemType = itemToShare?.item.isFolder ? 'folder' : 'file';
+    const itemId = itemToShare?.item.uuid ?? '';
+
+    const isItemNotSharedYet =
+      !isAdvanchedShareItem(itemToShare?.item) && !itemToShare.item.sharings?.length && !sharingMeta;
```

Comment:
Drive items not have `encryptionKey`. Will it work correctly in the Drive section? Has the property been added to return it in the Drive items to server-wip in this task?

Reply:
To access to itemToShare.item.sharings without Typescript complaining, I needed to narrow the type. So, this is the flow of this condition:

1. If `isAdvanchedShareItem` is false, it means it is not shared yet. This also means that the item comes from the **Drive** section.
2. So, we have a ! before the previous condition, now we check ` itemToShare.item.sharings?.length` (we know this comes from Drive already), if is other than 0, then the item was shared. 
3. If no, we check `SharingMeta` , which should be populated already by `loadShareInfo`. If this is also false, then the item has not been shared.

Reply:
Drive was my main section for testing on this case as we had a few unhandled cases when the modal is open from that section. Hope this fixes unhandled cases we have on production while also saves us a few requests to the backend haha

Reply:
ah excuse me, the conditional is checking if it fulfils the three conditions to affirm that it is not shared. I had read it wrong 🙏